server/src/main/java/com/linecorp/centraldogma/server/internal/replication/ReplicationLog.java

```diff
@@ -58,6 +60,9 @@ private static <T> T deserializeResult(
             result = null;
         }
 
+        if (command.type() == CommandType.FORCE_PUSH) {
+            command = ((ForcePushCommand<T>) command).delegate();
```

Comment:
Note: Understood that although it seems like the command is overriden, actually only the `resultType` is changed.

Perhaps modifying to something like the following may be more clear:
```java
final Class<T> resultType;
if (command.type() == CommandType.FORCE_PUSH) {
    resultType = (Class<T>) ((ForcePushCommand<T>) command).delegate().type().resultType();
} else {
    resultType = Util.unsafeCast(command.type().resultType());
}
```